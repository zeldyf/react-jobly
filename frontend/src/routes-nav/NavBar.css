.NavBar {
  background-color: white;
  border-bottom: 2px solid lightgray;
}

.nav-link.active {
  font-weight: bold;
  color: black;
}
