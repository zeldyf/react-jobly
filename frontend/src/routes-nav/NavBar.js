import React, { useContext } from "react";
import { NavLink, Link } from "react-router-dom";

import UserContext from "../UserContext";
import "./NavBar.css";

/** Navigation bar for site. Shows up on every page.
 *
 * When user is logged in, shows links to main areas of site. When not,
 * shows link to Login and Signup forms.
 *
 * Rendered by App.
 */

const NavBar = ({ logout }) => {
  const { currentUser } = useContext(UserContext);
  console.debug("NavBar", "currentUser=", currentUser);

  function loggedInNav() {
    return (
      <ul className="navbar-nav ml-auto">
        <li className="nav-item mr-4">
          <NavLink className="nav-link" to="/companies">
            Companies
          </NavLink>
        </li>
        <li className="nav-item mr-4">
          <NavLink className="nav-link" to="/jobs">
            Jobs
          </NavLink>
        </li>
        <li className="nav-item mr-4">
          <NavLink className="nav-link" to="/profile">
            Profile
          </NavLink>
        </li>
        <li className="nav-item">
          <Link className="nav-link" to="/" onClick={logout}>
            Log out {currentUser.first_name || currentUser.username}
          </Link>
        </li>
      </ul>
    );
  }

  function loggedOutNav() {
    return (
      <ul className="navbar-nav ml-auto">
        <li className="nav-item mr-4">
          <NavLink className="nav-link" to="/login">
            Login
          </NavLink>
        </li>
        <li className="nav-item mr-4">
          <NavLink className="nav-link" to="/signup">
            Sign Up
          </NavLink>
        </li>
      </ul>
    );
  }

  return (
    <nav className="NavBar navbar navbar-expand-md mb-5">
      <Link className="navbar-brand" to="/">
        Jobly
      </Link>
      {currentUser ? loggedInNav() : loggedOutNav()}
    </nav>
  );
};

export default NavBar;
