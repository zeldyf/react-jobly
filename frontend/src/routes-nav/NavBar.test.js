import React from 'react';
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import UserContext from "../UserContext";
import NavBar from "./NavBar";

test("renders without crashing", () => {
  const { getByRole } = render(
    <MemoryRouter>
      <UserContext.Provider value={{ currentUser: null }}>
        <NavBar />
      </UserContext.Provider>
    </MemoryRouter>
  );
  const navBar = getByRole("navigation");
  expect(navBar).toBeInTheDocument();
});

test("matches snapshot for logged out user", () => {
  const { container } = render(
    <MemoryRouter>
      <UserContext.Provider value={{ currentUser: null }}>
        <NavBar />
      </UserContext.Provider>
    </MemoryRouter>
  );
  expect(container).toMatchSnapshot();
});

test("matches snapshot for logged in user", () => {
  const { container } = render(
    <MemoryRouter>
      <UserContext.Provider value={{ currentUser: { username: "testuser" } }}>
        <NavBar />
      </UserContext.Provider>
    </MemoryRouter>
  );
  expect(container).toMatchSnapshot();
});
