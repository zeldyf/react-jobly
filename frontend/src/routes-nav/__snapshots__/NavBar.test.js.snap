// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot for logged in user 1`] = `
<div>
  <nav
    class="NavBar navbar navbar-expand-md mb-5"
  >
    <a
      class="navbar-brand"
      href="/"
    >
      Jobly
    </a>
    <ul
      class="navbar-nav ml-auto"
    >
      <li
        class="nav-item mr-4"
      >
        <a
          class="nav-link"
          href="/companies"
        >
          Companies
        </a>
      </li>
      <li
        class="nav-item mr-4"
      >
        <a
          class="nav-link"
          href="/jobs"
        >
          Jobs
        </a>
      </li>
      <li
        class="nav-item mr-4"
      >
        <a
          class="nav-link"
          href="/profile"
        >
          Profile
        </a>
      </li>
      <li
        class="nav-item"
      >
        <a
          class="nav-link"
          href="/"
        >
          Log out 
          testuser
        </a>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`matches snapshot for logged out user 1`] = `
<div>
  <nav
    class="NavBar navbar navbar-expand-md mb-5"
  >
    <a
      class="navbar-brand"
      href="/"
    >
      Jobly
    </a>
    <ul
      class="navbar-nav ml-auto"
    >
      <li
        class="nav-item mr-4"
      >
        <a
          class="nav-link"
          href="/login"
        >
          Login
        </a>
      </li>
      <li
        class="nav-item mr-4"
      >
        <a
          class="nav-link"
          href="/signup"
        >
          Sign Up
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
